File: modules/file-transfer-core/src/main/java/com/shterneregen/securelan/filetransfer/service/FileTransferAcceptanceHandler.java
package com.shterneregen.securelan.filetransfer.service;

import com.shterneregen.securelan.filetransfer.protocol.FileTransferMetadata;

@FunctionalInterface
public interface FileTransferAcceptanceHandler {
    boolean accept(FileTransferMetadata metadata, String remoteAddress);

    static FileTransferAcceptanceHandler acceptAll() {
        return (metadata, remoteAddress) -> true;
    }
}

File: modules/file-transfer-core/src/main/kotlin/com/shterneregen/securelan/filetransfer/service/FileTransferAcceptanceHandler.kt
package com.shterneregen.securelan.filetransfer.service

import com.shterneregen.securelan.filetransfer.protocol.FileTransferMetadata

fun interface FileTransferAcceptanceHandler {
    fun accept(metadata: FileTransferMetadata, remoteAddress: String): Boolean

    companion object {
        @JvmStatic
        fun acceptAll(): FileTransferAcceptanceHandler = FileTransferAcceptanceHandler { _, _ -> true }
    }
}
